### PR TITLE
[IMP] add product options from php serialize strings in sale line

### PR DESCRIPTION
**Pull request by bealdav**
Here is the type of format
opt

### CODE REVIEW

File: magentoerpconnect/sale.py [hunk 1]
```diff
@@ -1056,22 +1065,23 @@ def product_id(self, record):
...
+                bundle_select_unseria = phpserialize.loads(
+                    prd_opts['bundle_selection_attributes'].encode('utf-8'),
+                    decode_strings=True)
+                name += ': %s' % bundle_select_unseria['option_label']
+            # the bundle
+            if 'options' in record['product_options']:
+                options = [elm['label'] + ': ' + elm['print_value']
+                           for elm in
+                           record['product_options']['options'].values()]
+                name += '\nOptions:\n  - ' + '\n  - '.join(options)
```
Comment (guewen) [inline magentoerpconnect/sale.py]: What about translations?

**Comment by guewen** [comment]
Thanks for the patch.

However, I don't want to add a hard dependency for that since it is not used by many and could be an issue for existing installation.

I propose to replace the import by:

```
try:
    import phpserialize
except ImportError:
    phpserialize = None
```

And to check if phpserialize is not None before using it.
This optional dependency being explained in the readme.

**Comment by chafique-delli** [comment]
Effectively as told 'Guewen', I think that it would be better to have an optionnal dependency on 'phpserialize'.
May be an option to check in the backend ?

**Comment by bealdav** [comment]
HI @guewen, what do you think about @chafique-delli suggestion ? What should be the default behavior ?

**Comment by guewen** [comment]
> HI @guewen, what do you think about @chafique-delli suggestion ? What should be the default behavior ?

Hi,

No point to have an option in the backend. If the library is there, use it, otherwise just ignore the options.

**Comment by bealdav** [comment]
Hi @guewen, thanks but where I put options from previous mapping in case of none ?

**Comment by guewen** [comment]
> Hi @guewen, thanks but where I put options from previous mapping in case of none ?

I noticed that the previous mapping was writing the options in a field `notes` which does not exist. Possibly we can just keep the previous mapping in `notes` in v7 so if someone use it in that way it won't change the behavior.

**Comment by bealdav** [comment]
@guewen I fixed it but with by resetting my commit to keep old code. 
This first commit is without phpserialize, if ok, I'll modify .travis.yml to add this lib

File: magentoerpconnect/sale.py [hunk 2]
```diff
@@ -521,6 +534,18 @@ def _clean_magento_items(self, resource):
 
         # Group the childs with their parent
         for item in resource['items']:
+            if phpserialize is None:
+                _logger.info(PHPSERIALIZE_INFO)
```
Comment (guewen) [inline magentoerpconnect/sale.py]: This log will be logged at each import, I think it would be better to log it once in the `ImportError`.

**Comment by bealdav** [comment]
@guewen I suggest to add an item in 'Technical points' to alert of phpserialize feature

**Comment by guewen** [comment]
> @guewen I suggest to add an item in 'Technical points' to alert of phpserialize feature

Yep

**Comment by guewen** [comment]
Did you see the error in the tests?

```
2015-04-21 12:41:27,412 24044 TEST openerp_test openerp.modules.module: Traceback (most recent call last):
2015-04-21 12:41:27,412 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/tests/test_sale_order.py", line 112, in test_copy_quotation_delay_export_state
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `     binding = self._import_sale_order(900000691)
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/tests/test_sale_order.py", line 44, in _import_sale_order
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `     backend_id, increment_id)
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 378, in import_record
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `     importer.run(magento_id, force=force)
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/unit/import_synchronizer.py", line 206, in run
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `     self.magento_record = self._get_magento_data()
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/sale.py", line 688, in _get_magento_data
2015-04-21 12:41:27,413 24044 TEST openerp_test openerp.modules.module: `     record = self._clean_magento_items(record)
2015-04-21 12:41:27,414 24044 TEST openerp_test openerp.modules.module: `   File "/home/travis/build/OCA/connector-magento/magentoerpconnect/sale.py", line 545, in _clean_magento_items
2015-04-21 12:41:27,414 24044 TEST openerp_test openerp.modules.module: `     item[key].encode('utf-8'), decode_strings=True)
2015-04-21 12:41:27,414 24044 TEST openerp_test openerp.modules.module: ` AttributeError: 'dict' object has no attribute 'encode'
```

**Comment by pedrobaeza** [comment]
This one has conflicts and very old, so I'm closing it. If there's still interest, feel free to reopen it fixing the problems.

**Pull request closed without merging**